stingray/crossspectrum.py

```diff
@@ -433,10 +468,8 @@ def coherence(self):
         """
         # this computes the averaged power spectrum, but using the
         # cross spectrum code to avoid circular imports
-        ps1 = Crossspectrum(self.lc1, self.lc1)
-        ps2 = Crossspectrum(self.lc2, self.lc2)
 
-        return self.unnorm_power.real/(ps1.unnorm_power.real * ps2.unnorm_power.real)
+        return self.unnorm_power.real/(self.pds1.power.real * self.pds2.power.real)
```

Comment:
I don't know much about coherence, but you changed `unnorm_power`to (normalized) `power`, is this intentional? Was it a bug before? (the same comment for lines 745-746) :thinking: